django_ipam/base/models.py

```diff
@@ -1,28 +1,59 @@
+from ipaddress import ip_address, ip_network
+
 import swapper
+from django.conf import settings
+from django.core.exceptions import ValidationError
 from django.db import models
+from django.utils.translation import gettext_lazy as _
 from openwisp_utils.base import TimeStampedEditableModel
 
+from .fields import NetworkField
+
 
 class AbstractSubnet(TimeStampedEditableModel):
-    name = models.CharField(max_length=100, blank=True)  # Add a default value
+    subnet = NetworkField(default="10.0.0.0/24", db_index=True)
     description = models.CharField(max_length=100, blank=True)
+    master_subnet = models.ForeignKey('self', on_delete=models.CASCADE,
+                                      blank=True, null=True,
+                                      related_name="child_subnets")
 
     class Meta:
         abstract = True
+        indexes = [
+            models.Index(fields=['subnet'], name='subnet_idx')
+        ]
 
     def __str__(self):
-        return self.name
+        return str(self.subnet)
 
+    def clean(self):
+        for subnet in swapper.load_model("django_ipam", "Subnet").objects.filter().values():
+            if ip_network(self.subnet).overlaps(subnet["subnet"]):
+                raise ValidationError({'subnet': _('Subnet overlaps with %s') % (subnet["subnet"])})
 
-class AbstractIPAddress(TimeStampedEditableModel):
+
+class AbstractIpAddress(TimeStampedEditableModel):
     ip_address = models.GenericIPAddressField()
     description = models.CharField(max_length=100, blank=True)
     subnet = models.ForeignKey(swapper.get_model_name("django_ipam", "Subnet"),
                                on_delete=models.CASCADE,
                                blank=True, null=True)
+    owner = models.ForeignKey(settings.AUTH_USER_MODEL,
+                              on_delete=models.CASCADE,
+                              blank=True,
+                              null=True)
 
     class Meta:
         abstract = True
 
     def __str__(self):
         return self.ip_address
+
+    def clean(self):
+        if self.ip_address:
+            if self.subnet:
```

Comment:
if this field becomes mandatory as I think it should you can remove this if as well